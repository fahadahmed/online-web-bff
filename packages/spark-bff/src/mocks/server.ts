import { setupServer } from 'msw/node';
import a2pCustomerHandlers from 'datasources/a2p/customer/mocks/handlers';
import accessHandlers from 'datasources/user/access/mocks/handlers';
import accountSummaryHandlers from 'datasources/customer/accountsService/mocks/handlers';
import addressHandlers from 'datasources/location/address/mocks/handlers';
import airpointsHandlers from 'datasources/airpoints/mocks/handlers';
import availableNumbers from 'datasources/shop/availableNumbers/mocks/handlers';
import balanceServiceHandlers from 'datasources/finance/balanceService/mocks/handlers';
import billPaymentHandlers from 'datasources/finance/paymentService/billPayment/mocks/handlers';
import cartServicesHandlers from 'datasources/shop/cart/mocks/handlers';
import checkoutServicesHandlers from 'datasources/shop/checkout/mocks/handlers';
import connectionDiagnosticsHandlers from 'datasources/line/assure/connectionDiagnostics/mocks/handlers';
import connectionPromiseHandler from 'datasources/line/assure/connectionPromise/mocks/handlers';
import contentServiceHandlers from 'datasources/utility/content/mocks/handlers';
import digitalBillHandlers from 'datasources/finance/digitalBill/mocks/handlers';
import identityServiceHandlers from 'datasources/user/identityService/mocks/handlers';
import lineConfiguarationHandlers from 'datasources/line/lineConfiguration/mocks/handlers';
import lineDetailServiceHandlers from 'datasources/line/lineDetails/mocks/handlers';
import lineDetailsHandlers from 'datasources/line/lineSummary/mocks/handlers';
import lineSpendHistoryHandlers from 'datasources/line/spend/mocks/handlers';
import lineSubscriptionsService from 'datasources/line/lineSubscriptionsService/mocks/handlers';
import lineUsageDetailsMocks from 'datasources/line/usage/usageDetailsService/mocks';
import lineUsageHistoryMocks from 'datasources/line/usage/usageHistoryService/mocks';
import lineUsageMocks from 'datasources/line/usage/mocks/handlers';
import lineUsageTransactionMocks from 'datasources/line/usage/usageTransactionsService/mocks';
import networkSettingsHandlers from 'datasources/technology/networkSettingsService/mocks/handlers';
import orderHandlers from 'datasources/shop/order/mocks/handlers';
import paymentServiceHandlers from 'datasources/finance/paymentService/mocks/handlers';
import pointOfAddressHandlers from 'datasources/location/pointOfInterest/mocks/handlers';
import pointOfOutageHandlers from 'datasources/location/pointOfInterest/outagesPointOfInterest/mocks/handlers';
import preferenceHandlers from 'datasources/user/preference/mocks/handlers';
import prepaidBalanceServiceHandlers from 'datasources/finance/prepaidBalanceService/mocks/handlers';
import productDetailServiceHandlers from 'datasources/product/details/mocks/handlers';
import productFiltersServiceHandlers from 'datasources/product/filters/mocks/handlers';
import productOffersHandlers from 'datasources/product/offers/mocks/handlers';
import productPromotionsMPDServiceHandlers from 'datasources/product/promotions/mpd/mocks/handlers';
import profileServiceHandlers from 'datasources/marketing/profileService/mocks/handlers';
import resourceHandlers from 'datasources/utility/resource/mocks/handlers';
import roamingRatesHandlers from 'datasources/product/roamingRates/mocks/handlers';
import subscriptionActivationHandlers from 'datasources/line/subscriptionActivationService/mocks/handlers';
import topupHandlers from 'datasources/finance/paymentService/topup/mocks/handlers';
import transactionsListHandlers from 'datasources/line/transactionsService/mocks/handlers';
import troubleshootHandlers from 'datasources/line/assure/connectionTroubleshooter/mocks/handlers';
import userInfoHandlers from 'datasources/user/userInfo/mocks/handlers';
import walletServiceHandlers from 'datasources/finance/walletService/mocks/handlers';

const handlers = [
  ...a2pCustomerHandlers,
  ...accessHandlers,
  ...accountSummaryHandlers,
  ...addressHandlers,
  ...availableNumbers,
  ...airpointsHandlers,
  ...balanceServiceHandlers,
  ...billPaymentHandlers,
  ...cartServicesHandlers,
  ...checkoutServicesHandlers,
  ...connectionDiagnosticsHandlers,
  ...connectionPromiseHandler,
  ...contentServiceHandlers,
  ...digitalBillHandlers,
  ...identityServiceHandlers,
  ...lineConfiguarationHandlers,
  ...lineDetailServiceHandlers,
  ...lineDetailsHandlers,
  ...lineSpendHistoryHandlers,
  ...lineSubscriptionsService,
  ...lineUsageDetailsMocks,
  ...lineUsageHistoryMocks,
  ...lineUsageMocks,
  ...lineUsageTransactionMocks,
  ...networkSettingsHandlers,
  ...orderHandlers,
  ...paymentServiceHandlers,
  ...pointOfAddressHandlers,
  ...pointOfOutageHandlers,
  ...preferenceHandlers,
  ...prepaidBalanceServiceHandlers,
  ...productDetailServiceHandlers,
  ...productFiltersServiceHandlers,
  ...productOffersHandlers,
  ...productPromotionsMPDServiceHandlers,
  ...profileServiceHandlers,
  ...resourceHandlers,
  ...roamingRatesHandlers,
  ...subscriptionActivationHandlers,
  ...topupHandlers,
  ...transactionsListHandlers,
  ...troubleshootHandlers,
  ...userInfoHandlers,
  ...walletServiceHandlers,
];

const server = setupServer(...handlers);
export default server;
