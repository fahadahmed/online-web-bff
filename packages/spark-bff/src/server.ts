import { ApolloServer } from 'apollo-server-fastify';
import {
  ApolloServerPluginLandingPageDisabled,
  ApolloServerPluginLandingPageGraphQLPlayground,
} from 'apollo-server-core';
import { logger } from 'apolloLogger';
import context from 'context';
import dataSources from 'datasources';
import { ApolloLoggerPlugin, ApplicationLogEvent } from 'logging';
import schema from 'schema';
import { formatResponse } from 'utils/formatResponse';
import { isDevelopmentEnvironment } from 'utils/isDevelopmentEnvironment';

process.on('uncaughtException', (err) => {
  logger.error(new ApplicationLogEvent({ message: err.message, context: {} }));
  throw err;
});

const apolloServer = new ApolloServer({
  schema,
  dataSources,
  formatResponse,
  logger,
  context,
  introspection: isDevelopmentEnvironment(),
  plugins: [
    new ApolloLoggerPlugin(
      `BFF Server Started. BUILD_ID: ${process.env.BUILD_BUILDID}`,
    ),
    isDevelopmentEnvironment()
      ? ApolloServerPluginLandingPageGraphQLPlayground()
      : ApolloServerPluginLandingPageDisabled(),
  ],
});

export default apolloServer;
