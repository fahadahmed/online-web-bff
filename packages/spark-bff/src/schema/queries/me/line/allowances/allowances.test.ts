import { server } from 'test/server';
import lineDetailsMockHandlers from 'datasources/line/lineDetails/mocks/handlers';
import { createTestClient, gql } from 'test/apolloTestUtils';

beforeEach(() => {
  server.use(...lineDetailsMockHandlers);
});

test('returns allowances from a given line number', async () => {
  const { query } = createTestClient();
  const res = await query({
    query: gql`
      query {
        me {
          line(lineNumber: "1234567") {
            allowances {
              dataStackAllowances {
                capLimit
              }
              entitlements {
                description
                detailedDescription
                displayPriority
                id
                name
                quantity
                subtype
                throttleMessage
                unlimited
                usageType
                entitlementQuantityUnit
              }
              extras {
                allowedCount
                availableCount
                billingFrequencyCode
                billingFrequencyName
                contractRenewalType
                contractTermFrequency
                contractTermNumber
                countries {
                  name
                }
                displayPriority
                entitlements {
                  description
                }
                renewable
                offerId
                periodEndDate
                prepaidBasePlanValuePack
                price
                priceType
                productId
                productInstanceId
                productName
                renewalFrequency
                renewalState
                status
                subscriptionTermName
                subtype
                type
                usageType
              }
              rolloverDetails {
                rolloverAllowances {
                  remainingAmount
                }
              }
              sharePlanDetails {
                groupId
              }
              speedAllowances {
                status
              }
            }
          }
        }
      }
    `,
  });
  expect(res.errors).toBeUndefined();
  expect(res.data!.me.line.allowances).toMatchSnapshot();
});
