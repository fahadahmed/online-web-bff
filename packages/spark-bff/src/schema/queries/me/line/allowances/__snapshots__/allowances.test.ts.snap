// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`returns allowances from a given line number 1`] = `
Object {
  "dataStackAllowances": Array [],
  "entitlements": Array [
    Object {
      "description": "120GB included data",
      "detailedDescription": null,
      "displayPriority": null,
      "entitlementQuantityUnit": "GB",
      "id": "29357_M_DATA(WB)_120GB",
      "name": "Data",
      "quantity": 120,
      "subtype": null,
      "throttleMessage": null,
      "unlimited": false,
      "usageType": "Data",
    },
    Object {
      "description": "Data Lift",
      "detailedDescription": null,
      "displayPriority": null,
      "entitlementQuantityUnit": "GB",
      "id": "24226_M_DATA-WB-TIER_1_10GB",
      "name": "Data",
      "quantity": 10,
      "subtype": null,
      "throttleMessage": null,
      "unlimited": false,
      "usageType": "Data",
    },
    Object {
      "description": "500 minutes included voice",
      "detailedDescription": null,
      "displayPriority": null,
      "entitlementQuantityUnit": "MIN",
      "id": "29357_M_VOICE(WB)_500MIN",
      "name": "Voice",
      "quantity": 500,
      "subtype": null,
      "throttleMessage": null,
      "unlimited": false,
      "usageType": "Voice",
    },
    Object {
      "description": "unlimited texts",
      "detailedDescription": null,
      "displayPriority": null,
      "entitlementQuantityUnit": "TEXT",
      "id": "29357_M_TEXT(WB)_UNLIMITED",
      "name": "Text",
      "quantity": null,
      "subtype": null,
      "throttleMessage": null,
      "unlimited": true,
      "usageType": "Text",
    },
  ],
  "extras": Array [
    Object {
      "allowedCount": null,
      "availableCount": null,
      "billingFrequencyCode": "monthly",
      "billingFrequencyName": "Monthly",
      "contractRenewalType": "auto_renewing",
      "contractTermFrequency": "monthly",
      "contractTermNumber": 0,
      "countries": Array [],
      "displayPriority": null,
      "entitlements": Array [],
      "offerId": "legacy050114",
      "periodEndDate": null,
      "prepaidBasePlanValuePack": false,
      "price": 8,
      "priceType": "GST_EXEMPT",
      "productId": "VoFFeaturePack2003",
      "productInstanceId": "1-1MSNNLMR",
      "productName": "Connector Feature Pack",
      "renewable": true,
      "renewalFrequency": "Monthly",
      "renewalState": "OFF",
      "status": "ACTIVE",
      "subscriptionTermName": "1 Month contract - monthly billing",
      "subtype": "DOMESTIC",
      "type": "ServiceAddOn",
      "usageType": "Voice",
    },
  ],
  "rolloverDetails": null,
  "sharePlanDetails": null,
  "speedAllowances": null,
}
`;
